marconi-sidechain/doc/API.adoc

```diff
@@ -147,745 +112,585 @@ Taking the chainpoint before ensure that we have consistent infomation across al
       "type": "string"
     },
-    "method": {
-      "type": "string"
+    "blockNo": {
+      "minimum": 0,
+      "type": "integer"
     },
-    "params": {
-      "type": "integer",
+    "epochNo": {
       "minimum": 0,
-      "description": "Epoch number"
+      "type": "integer"
     },
-    "id": {
+    "slotNo": {
+      "minimum": 0,
       "type": "integer"
+    },
+    "nonce": {
+      "pattern": "^[0-9a-f]{64}$",
+      "type": "string"
     }
   },
   "required": [
-    "jsonrpc",
-    "method",
-    "params",
-    "id"
+    "blockHeaderHash",
+    "blockNo",
+    "epochNo",
+    "nonce",
+    "slotNo"
   ]
 }
 ```
 ====
 
-.HTTP Response body
+.JSON-RPC error data object
 [%collapsible]
 ====
 ```JSON
 {
-  "$schema": "http://json-schema.org/draft-04/schema#",
-  "type": "object",
   "oneOf": [
     {
-      "type": "object",
-      "properties": {
-        "id": {
-          "type": "integer"
-        },
-        "jsonrpc": {
-          "type": "string"
-        },
-        "result": {
-          "type": "object",
-          "properties": {
-            "blockHeaderHash": {
-              "pattern": "^[0-9a-f]{64}$",
-              "type": "string"
-            },
-            "blockNo": {
-              "minimum": 0,
-              "type": "integer"
-            },
-            "epochNo": {
-              "minimum": 0,
-              "type": "integer"
-            },
-            "slotNo": {
-              "minimum": 0,
-              "type": "integer"
-            },
-            "nonce": {
-              "pattern": "^[0-9a-f]{64}$",
-              "type": "string"
-            }
-          },
-          "required": [
-            "blockHeaderHash",
-            "blockNo",
-            "epochNo",
-            "nonce",
-            "slotNo"
-          ]
-        }
-      },
-      "required": [
-        "id",
-        "jsonrpc",
-        "result"
-      ]
+      "const": "The 'epochNo' param value must be a natural number."
     },
     {
-      "type": "object",
-      "properties": {
-        "id": {
-          "type": "integer"
-        },
-        "jsonrpc": {
-          "type": "string"
-        },
-        "error": {
-          "type": "object",
-          "properties": {
-            "code": {
-              "type": "integer"
-            },
-            "message": {
-              "type": "integer"
-            },
-            "data": {
-              "type": "integer"
-            }
-          },
-          "required": [
-            "code",
-            "message"
-          ]
-        }
-      },
-      "required": [
-        "id",
-        "jsonrpc",
-        "error"
-      ]
+      "const": "The 'epochNo' param value must belong in an era that has nonces (Byron era doesn't have nonces)."
     }
   ]
 }
 ```
 ====
 
+
+=== getActiveStakePoolDelegationByEpoch
+
+Retrieves the active stake pool delegation per epoch.
+
 ==== Pre-conditions
 
-TBD
+* The `epochNo` param value must be a natural number.
```

Comment:
I don't remember, do we track it for all `epochNo`? If no, shouldn't this restriction be clearly stated?

Reply:
The other error message below basically says that: "The 'epochNo' param value must belong in an era that has nonces (Byron era doesn't have nonces)."

Reply:
Good point. Added it as follows:

```
The `nonce` result value must be in an era that has epoch nonces (for example, Byron era doesn't have nonces).
```